Design/MoreFlexibleILiveListener.md

```diff
@@ -63,34 +64,89 @@ The general problem we are trying to solve is that the instrument is assumed to
 
 
-### Option 3. Overload the `LiveListenerFactory` connect
+### Option 2: Overload the `LiveListenerFactory` connect
 
 Currently `LiveListenerFactory::create` as called by `LiveDataAlgorithm` will:
 
 1. Load the Instrument and then InstrumentInfo for the specified *Instrument* name argument. All taken from the Facility file.
-1. Extract the concrete ILiveListener from the above and create an instance of one
-1. Connect it from the address specified above
+1. Extract the concrete ILiveListener from the above and create an instance of one.
+1. Connect it from the address specified in the Facilities.xml.
 
-The only thing that any `create` method should do is Create the relevant object, and we should have a way of providing the ILiveListenter class name ourselves, and connecting the product ourselves. 
+The only thing that any `create` method should do is create the relevant object, and we should have a way of providing the ILiveListener class name ourselves, and connecting the product ourselves.
 
-The completed solution would allow users to specify the stuff usually loaded from the facilities.xml file directly in `StartLiveData` namely the address, ILiveListener name and component name. If provided the code would not try to then fetch that information from the instrument.
+The completed solution would allow users to specify the stuff usually loaded from the facilities.xml file directly in `StartLiveData`. Namely, the address, ILiveListener name and component name. If provided, the code would not try to then fetch that information from the instrument.
 
 **In this solution users would be able to use `StartLiveData` and completely bypass the Facilities.xml file.** Again, this can be added in parallel to the above fixes.
+
+
+Proposed Solution
+-----------------
+
+#### Requirement 1
+
+Dynamic properties for listeners were already supported previously, but this feature was removed in [PR #234](https://github.com/mantidproject/mantid/pull/234) in response to [Trac #11059](http://trac.mantidproject.org/mantid/ticket/11059) because it interfered with help generation and Python API support.
+
+Reverting this PR will satisfy this requirement, but we will need to deal with the help and Python issues in a different way. This is beyond the scope of this design and can be handled separately.
```

Comment:
@martyngigg. While i think this change set should be reverted. I think that @MikeHart85  has a good point about the way that the python functional syntax for the `StartLiveData` is created. Do you have any suggestions to get around this?